### DISCUSSION TITLE
examples not working for me

### DISCUSSION THREAD

**Discussion by SpieringsAE** [christoph2/pyxcp#100] [Q&A]
Hello,

I want to use this library to read information from a running program via xcp.
And as there is not a lot of documentation yet i thought let's try some of the examples and see what happens.

However they don't seem to be working for me, most give this error: TRANSPORT must be specified in config!
Which comes from config.py, but all of the other conf files seem to contain TRANSPORT fields that have entries in them.

and when i run xcptest.py it complains about the entries of the Eth() method:

`
Exception has occurred: TypeError
__init__() got an unexpected keyword argument 'port'
  File "/etc/pyxcptest.py", line 92, in cstest
    tr = transport.Eth("localhost", port=5555, protocol="UDP", loglevel="WARN")  # "DEBUG"
  File "/etc/pyxcptest.py", line 590, in <module>
    cstest()

`

I think I would like to use the Ethernet protocol, as the program I'm monitoring is running locally on the same controller as the python script I'm trying to make.

But i also have a general question, I'm used to having to load an A2L file to use xcp, but i haven't seen anything related to reading one in the examples. Does the library get that information some other way somehow?

I must admit that I am quite new to stuff this indepth in python.

**Comment by christoph2** [comment]
`File "/etc/pyxcptest.py", line 92, in cstest` do you mean examples/xcptest.py ?

This is a legacy script and indeed not working (I've just removed it).

As a starting point you should use:
```shell
python xcphello.py -c conf_eth.toml
```
There are various example configurations in  .json and .toml format.

[Here](https://github.com/christoph2/pyxcp/blob/master/docs/configuration.rst) you find the configuration documentation.

---

For A2L action in general use [pya2ldb](https://github.com/christoph2/pyA2L).
Both projects are mainly mechanism-not-policy, i.e. rather low-level, for this reason I've created
[asamint](https://github.com/christoph2/asamint) -- there's still much work to do, but there are some working [examples](https://github.com/christoph2/asamint/tree/master/asamint/examples), like creating calibration data files (ASAM CDF and Damos DCM).
Documentation will follow (really...)

**Reply by SpieringsAE** [reply]
Thanks for your quick reply!

I really appreciate the work you are doing here, I will try out your feedback.

I tend to be behind on the documentation part aswell so i understand hahaha.

**Comment by bsr656** [comment]
Hi , I am trying to run xcphello.py example script and I have the following error. Could you please help me?
2025-03-28 13:47:22 INFO     XCPonEth - Connected to: 10.1.0.32:5555  local        eth.py:103
                             address: 192.168.1.12:5556 [AF_INET][SOCK_DGRAM]
2025-03-28 13:47:27 WARNING  XcpTimeoutError -- Service: 'connect'        errorhandler.py:371
2025-03-28 13:47:32 WARNING  XcpTimeoutError -- Service: 'connect'        errorhandler.py:371
2025-03-28 13:47:37 WARNING  XcpTimeoutError -- Service: 'connect'        errorhandler.py:371
2025-03-28 13:47:42 WARNING  XcpTimeoutError -- Service: 'connect'        errorhandler.py:371
                    ERROR    Traceback (most recent call last):                 master.py:129
                               File
                             "../python3.10/site-pack
                             ages/pyxcp/master/errorhandler.py", line 364, in
                             __call__
                                 res = handler.execute()
                               File
                             "../python3.10/site-pack
                             ages/pyxcp/master/errorhandler.py", line 230, in
                             execute
                                 return self.func(self.instance,
                             *self.arguments.args, **self.arguments.kwargs)
                               File
                             "../python3.10/site-pack
                             ages/pyxcp/master/master.py", line 176, in connect
                                 response =
                             self.transport.request(types.Command.CONNECT, mode
                             & 0xFF)
                               File
                             "../python3.10/site-pack
                             ages/pyxcp/transport/base.py", line 260, in
                             request
                                 return self._request_internal(cmd, False,
                             *data)
                               File
                             "../python3.10/site-pack
                             ages/pyxcp/transport/base.py", line 246, in
                             _request_internal
                                 raise types.XcpTimeoutError(MSG) from None
                             pyxcp.types.XcpTimeoutError: Response timed out
                             (timeout=5.0s)

                             During handling of the above exception, another
                             exception occurred:

                             Traceback (most recent call last):
                               File
                             "/home/azure-agent/pipelines-agent/work/1/s/tools/
                             scripts/integration_test/automated_smoke_tests/tes
                             ts/examples/xcphello.py", line 29, in <module>
                                 x.connect()
                               File
                             "../python3.10/site-pack
                             ages/pyxcp/master/errorhandler.py", line 437, in
                             inner
                                 res = executor(inst, func, arguments)
                               File
                             "../python3.10/site-pack
                             ages/pyxcp/master/errorhandler.py", line 376, in
                             __call__
                                 raise XcpTimeoutError("Maximum CONNECT retries
                             reached.")
                             pyxcp.types.XcpTimeoutError: Maximum CONNECT
                             retries reached.
